src/plugins/mer/merhardwaredevicewizardpages.cpp

```diff
@@ -138,34 +139,40 @@ void MerHardwareDeviceWizardSelectionPage::handleTestConnectionClicked()
     sshParams.timeout = timeout();
     sshParams.authenticationType = SshConnectionParameters::AuthenticationTypeAll;
 
+    QString error;
     m_ui->connectionTestLabel->setText(tr("Connecting to machine %1 ...").arg(hostName()));
     m_ui->connectionTestLabel->setText(MerConnectionManager::testConnection(sshParams,
                 &m_connectionTestOk));
     if (!m_connectionTestOk)
         goto end;
 
     m_ui->connectionTestLabel->setText(tr("Detecting device properties..."));
-    m_architecture = detectArchitecture(sshParams, &m_connectionTestOk);
-    m_deviceName = detectDeviceName(sshParams, &m_connectionTestOk);
+    m_architecture = detectArchitecture(sshParams, &m_connectionTestOk, &error);
+    if (m_connectionTestOk) {
+        m_deviceName = detectDeviceName(sshParams, &m_connectionTestOk, &error);
+    }
     if (!m_connectionTestOk) {
         m_ui->connectionTestLabel->setText(tr("Could not autodetect device properties"));
+        QMessageBox::critical(this, tr("Device Error"), error);
```

Comment:
The "Device Error" message sounds a bit drastic :) How about "Connection test failed"?